server/chat/backend/agent/tools/flyio_tool.py

```
@@ -0,0 +1,80 @@
+        return bool(token_data and "api_token" in token_data)
+    except Exception:
+        return False
+
+
+class FlyioMetricsQueryArgs(BaseModel):
+    """Arguments for querying Fly.io Prometheus metrics."""
+    query: str = Field(description="PromQL query (e.g. 'fly_instance_up{app=\"myapp\"}', 'rate(fly_instance_cpu{app=\"myapp\"}[5m])')")
+    time: Optional[str] = Field(default=None, description="Evaluation time (RFC3339 or Unix timestamp). Defaults to now.")
+
+
+def query_flyio_metrics(query: str, time: Optional[str] = None, user_id: str = None, session_id: str = None) -> str:
```

Ruff (0.15.15)

[warning] PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)